docs/dapp/opl/router-protocol/pingpong-example.md

```diff
@@ -0,0 +1,351 @@
+                string memory asmAddress
+            ) public pure returns (bytes memory) {
+                bytes memory requestMetadata = abi.encodePacked(
+                destGasLimit,
+                destGasPrice,
+                ackGasLimit,
+                ackGasPrice,
+                relayerFees,
+                ackType,
+                isReadCall,
+                asmAddress
+                );
+                return requestMetadata;
+            }
+
+            /// @notice function to handle the cross-chain request received from some other chain.
+            /// @param packet the payload sent by the source chain contract when the request was created.
+            /// @param srcChainId chain ID of the source chain in string.
+            function iReceive(
+                string memory, //requestSender,
+                bytes memory packet,
+                string memory srcChainId
+            ) external returns (uint64, string memory) {
+                require(msg.sender == address(gatewayContract), "only gateway");
+                (uint64 requestId, string memory sampleStr) = abi.decode(
+                packet,
+                (uint64, string)
+                );
+                if (
+                keccak256(abi.encodePacked(sampleStr)) == keccak256(abi.encodePacked(""))
+                ) {
+                revert CustomError("String should not be empty");
+                }
+                pingFromSource[srcChainId][requestId] = sampleStr;
+
+                emit PingFromSource(srcChainId, requestId, sampleStr);
+
+                return (requestId, sampleStr);
+            }
+
+            /// @notice function to handle the acknowledgement received from the destination chain
+            /// back on the source chain.
+            /// @param requestIdentifier event nonce which is received when we create a cross-chain request
+            /// We can use it to keep a mapping of which nonces have been executed and which did not.
+            /// @param execFlag a boolean value suggesting whether the call was successfully
+            /// executed on the destination chain.
+            /// @param execData returning the data returned from the handleRequestFromSource
+            /// function of the destination chain.
+            function iAck(
+                uint256 requestIdentifier,
+                bool execFlag,
+                bytes memory execData
+            ) external {
+                (uint64 requestId, string memory ackMessage) = abi.decode(
+                execData,
+                (uint64, string)
+                );
+
+                ackFromDestination[requestId] = ackMessage;
+
+                emit ExecutionStatus(requestIdentifier, execFlag);
+                emit AckFromDestination(requestId, ackMessage);
+            }
+        }
+        ```
+  </details>
+
+### Key Contract Functions
+
+- `iPing`: Initiates the cross-chain message by calling Router's
+  `IGateway.iSend`.
```

Comment:
(unrelated) I do find the new backticked code to not look as nice.

<img width="668" alt="image" src="https://github.com/user-attachments/assets/5efa43e3-507c-484a-ab5e-15d77b51a1c5">